### ISSUE TITLE
[e10s] when clicking <select>s, the main window drops down.

### ISSUE THREAD

**Issue by suishouen** [johngraciliano/simplewhite#31]
Steps to reproduce:
1. Enable multi-process (e10s).
2. Open a window including <select>s.
   e.g. http://forums.mozillazine.org/viewforum.php?f=49&sid=aa15e37fb4e83690adb1d84becb68b24
3. Click <select>s.
   e.g. Mozilla Firefox
4. The main window drops down.

I'm using OS X Version 10.11.6.
I don't know this happens only on Mac or not.
![disable e10s](https://cloud.githubusercontent.com/assets/21173308/17854063/071392d4-68ab-11e6-9721-0a5007e3341a.png)

![enable e10s](https://cloud.githubusercontent.com/assets/21173308/17854072/1203f184-68ab-11e6-89f9-812899be04cf.png)

**Comment by johngraciliano** [comment]
I am unclear what is equivalent to <select> s in Windows 8.  It is very hard to try to reproduce and understand.  Your images do show a serious problem.  What version of Firefox is this?  Are there any other extensions installed?  I may need more information and perhaps even more help from you.

**Comment by suishouen** [comment]
Thanks for your reply.
1. I'm testing Firefox 51.0a1 with Simple White X v2.2b18.x1.22.
2. I'm testing with fresh profile with no other xtensions.
3. While clicking <select>, the window drops down as shown "Enable e10s".
4. Releasing <select>, the window lifts back as shown "Disable e10s".

**Comment by suishouen** [comment]
Additional Info:
I've just tested Firefox-50.0a2 (latest-mozilla-aurora) with fresh profile using Simple White X v2.2b18.x1.22.
And that's the same. This issue happens, too.

**Comment by suishouen** [comment]
1. I've also tested latest beta Firefox 49.0b5 with fresh profile.
2. Set "browser.tabs.remote.force-enable" to "true" for enableing multi-process (e10s).
3. Open http://forums.mozillazine.org/viewforum.php?f=49&sid=aa15e37fb4e83690adb1d84becb68b24.
4. I confirm this issue happens on Firefox 49.0b5, too.

**Comment by suishouen** [comment]
1. Now I've tested latest release Firefox 48.0.1.
2. Setting "about:config", and confirm Multiprocess Windows : 1/1 (Enabled by user) on "Troubleshooting Information" window.
3. That's all the same.
4. When enabling multiprocess windows (e10s), this issue happens Firefox 48, 49, 50 and 51 on Mac.

**Comment by suishouen** [comment]
When using default theme, this issue doesn't happen.
So with trial and error, I finally found the solution to fix this issue.

Here's the patch to fix this issue.

```
--- /whitefox/ostarget/darwin/global/menulist.css   2016-08-22 17:06:06.000000000 +0900
+++ /whitefox/ostarget/darwin/global/menulist.css   2016-08-24 14:25:13.000000000 +0900
@@ -8,11 +8,14 @@
 menulist {
   -moz-appearance: menulist;
   margin: 5px 2px 3px;
-  min-height: 20px !important;
   color: -moz-DialogText;
   text-shadow: none;
 }

+menulist:not([popuponly="true"]) {
+  min-height: 20px;
+}
+
 .menulist-label-box {
   -moz-appearance: menulist-text;
   -moz-box-align: center;
```

I hope you find it informative.
Best Regards,

**Comment by johngraciliano** [comment]
Sorry for my very late response, I am often away from the _World Wide Web_.  A have just noticed your great contribution, so I will need to act on it.  
P.S.  I am updating many of the global styles (`menulist.css` included).  Many of the styles sheets used are different for the **mac** platform but I have no idea how to get a hold of the default theme used for **OS X**.  I know how to grab it easily from the **Windows** code I use (inside `.ja` files) but that is it.  Any pointers are welcome.

**Comment by suishouen** [comment]
I haven't used **Windows**, so I don't know the following way **works or not**.
But please try.

Theme files exist in:
Firefox.app/Contents/Resources/omni.ja
Firefox.app/Contents/Resources/browser/omni.ja

For example Firefox 48.0.1
Open Terminal and run below:

``` #!/bin/bash
# Download and mount .dmg file
cd ~/Desktop
curl -O http://ftp.mozilla.org/pub/firefox/releases/48.0.1/mac/en-US/Firefox%2048.0.1.dmg
hdiutil mount Firefox%2048.0.1.dmg

# Make Directory omni1 & omni2
mkdir omni1 omni2

# Copy Theme Files to omni1 & omni2 on your Desktop
cp -R /Volumes/Firefox/Firefox.app/Contents/Resources/browser/omni.ja ~/Desktop/omni1
cp -R /Volumes/Firefox/Firefox.app/Contents/Resources/omni.ja ~/Desktop/omni2
cd ~/Desktop/omni1
unzip ~/Desktop/omni1/omni.ja
cd ~/Desktop/omni2
unzip ~/Desktop/omni2/omni.ja

# Delete .dmg file and unmount 
rm -r ~/Desktop/Firefox%2048.0.1.dmg
hdiutil unmount /Volumes/Firefox

exit
```

If you succeed, you can get files:
/chrome/skin/browser files at Desktop/omni1/chrome/browser/skin/classic/browser
/chrome/skin/devtools at  Desktop/omni1/chrome/devtools/skin
/chrome/global at Desktop/omni2/chrome/toolkit/skin/classic/global
/chrome/mozapps at Desktop/omni2/chrome/toolkit/skin/classic/mozapps

Latest beta, aurora and central
https://ftp.mozilla.org/pub/firefox/releases/49.0b7/mac/en-US/Firefox%2049.0b7.dmg
https://ftp.mozilla.org/pub/firefox/nightly/latest-mozilla-aurora/firefox-50.0a2.en-US.mac.dmg
https://ftp.mozilla.org/pub/firefox/nightly/latest-mozilla-central/firefox-51.0a1.en-US.mac.dmg

Best Regards.

**Comment by suishouen** [comment]
Maybe this is possible for Windows.

For example,Firefox 48.0.2
1. Download Source Files from:
https://ftp.mozilla.org/pub/firefox/releases/48.0.2/source/firefox-48.0.2.source.tar.xz
2. Decompress firefox-48.0.2.source.tar.xz
3. You can find files as below:
skin browser at firefox-48.0.2.source/firefox-48.0.2/browser/themes/osx
skin devtools at firefox-48.0.2.source/firefox-48.0.2/devtools/client/themes
skin global at firefox-48.0.2.source/firefox-48.0.2/toolkit/themes/osx/global
skin mozapps at firefox-48.0.2.source/firefox-48.0.2/toolkit/themes/osx/mozapps

In the same way Firefox 49.0b7:
https://ftp.mozilla.org/pub/firefox/releases/49.0b7/source/firefox-49.0b7.source.tar.xz

At least you can use this method to get source files of Firefox 48 and Firefox 49, I think.

**Comment by suishouen** [comment]
About Firefox 50.0a2 and Firefox 51.0a1, I can't find source file tarball.
So you have to get source files individually.

Firefox 50.0a2 (latest aurora)
skin browser at https://hg.mozilla.org/releases/mozilla-aurora/file/tip/browser/themes/osx
skin devtools at https://hg.mozilla.org/releases/mozilla-aurora/file/tip/devtools/client/themes
skin global at https://hg.mozilla.org/releases/mozilla-aurora/file/tip/toolkit/themes/osx/global
skin mozapps at https://hg.mozilla.org/releases/mozilla-aurora/file/tip/toolkit/themes/osx/mozapps

Firefox 51.0a1 (latest central)
skin browser at https://hg.mozilla.org/mozilla-central/file/tip/browser/themes/osx
skin devtools at https://hg.mozilla.org/mozilla-central/file/tip/devtools/client/themes
skin global at https://hg.mozilla.org/mozilla-central/file/tip/toolkit/themes/osx/global
skin mozapps at https://hg.mozilla.org/mozilla-central/file/tip/toolkit/themes/osx/mozapps

**Comment by suishouen** [comment]
Firefox 50 and 51 source tarball are here.

https://hg.mozilla.org/mozilla-central/archive/tip.zip
https://hg.mozilla.org/releases/mozilla-aurora/archive/tip.zip

**Comment by suishouen** [comment]
Sorry for needing additional files about skin mozapps:

skin mozapps at firefox-48.0.2.source/firefox-48.0.2/toolkit/themes/shared/mozapps
skin mozapps at firefox-49.0b7.source/firefox-49.0b7/toolkit/themes/shared/mozapps

skin mozapps at https://hg.mozilla.org/releases/mozilla-aurora/file/tip/toolkit/themes/shared/
skin mozapps at https://hg.mozilla.org/mozilla-central/file/tip/toolkit/themes/shared/

Best Regards.

**Comment by Lootyhoof** [comment]
For reference, it is possible to open .dmg files on non-Mac platforms with 7-Zip and similar:

[file].dmg/Firefox/Firefox.app/Contents/MacOS/omni.ja (for /global and /mozapps)
[file].dmg/Firefox/Firefox.app/Contents/MacOS/browser/omni.ja (for /browser)

**Comment by suishouen** [comment]
> For reference, it is possible to open .dmg files on non-Mac platforms with 7-Zip and similar:

.app files are application package files on **OS X**.
I'm not sure whether the following way is possible on **Windows**, too.

1) Open .dmg file and copy Firefox.app to your Desktop.
2) Move the file name of Firefox.app to Firefox. (delete .app suffix)
3) So you will be able to see a folder (named Firefox) and find contents of Firefox.app in it.

**Comment by johngraciliano** [comment]
When I open **Firefox 49.0b7.dmg** with PeaZip I can navigate to
Firefox/Firefox.app/Contents/MacOS/
However I cannot find omni.ja nor any usable structure.  I can only see a big file named XUL after other files and folders.  I am having a bit of trouble with the .tar.xz for size reasons.  Thank you for the pointers, I will see how far can I get since I am just starting.

**Comment by suishouen** [comment]
> Firefox/Firefox.app/Contents/MacOS/
> However I cannot find omni.ja nor any usable structure.

You will find two omni.ja files at:
・Firefox.app/Contents/Resources/browser/omni.ja
・Firefox.app/Contents/Resources/omni.ja

**Comment by johngraciliano** [comment]
Thank you.  That works 👍 
It may still be a week or two before I am done upgrading the theme properly for the mac  (especially because I cannot test so I have to be very careful).
I will to include your ( @suishouen ) correction much sooner than that.

**Comment by suishouen** [comment]
Thanks.
I confirmed and verified the fix with Version 2.2b18.x1.23.
Please close this issue.

**Comment by johngraciliano** [comment]
@suishouen: _Thank you_ so much for all your help, which went beyond _identifying_ and _solving_ this issue.
Please let me know about any other issue regarding **OS X**.  I am particularly concerned about issue #32 because I was hoping the smaller print in the suggestion box to be as large as the entry print in the location bar.
